specs/_features/maxeb_increase/capella.py

```diff
@@ -1627,6 +1652,83 @@ def process_pending_balance_deposits(state: BeaconState) -> None:
+    current_epoch = get_current_epoch(state)
+
+    # Resolve the target of consolidation
+    target_index = resolve_consolidated_to(state, pending_consolidation.target_index)
+    target_validator = state.validators[target_index]
+    source_validator = state.validators[pending_consolidation.source_index]
+
+    # Apply consolidation
+    source_validator.consolidated_to = target_index
+
+    # Don't move the balance and slash the target if the source was slashed
+    if is_slashed(source_validator):
+        if is_slashed_proposer(source_validator):
+            penalty = PROPOSER_EQUIVOCATION_PENALTY_FACTOR * EFFECTIVE_BALANCE_INCREMENT
+            decrease_balance(state, target_validator, penalty)
+            initiate_validator_exit(state, target_index)
+            target_validator.slashed = add_flag(target_validator.slashed, SLASHED_PROPOSER_FLAG_INDEX)
+
+        if is_slashed_attester(source_validator):
+            if is_attester_slashable_validator(target_validator, current_epoch):
+                slash_validator(state, target_index)
+
+        return
+
+    # Don't move the balance and slash the source if the target was slashed
+    if is_slashed(target_validator):
+        if is_slashed_proposer(target_validator):
+            penalty = PROPOSER_EQUIVOCATION_PENALTY_FACTOR * EFFECTIVE_BALANCE_INCREMENT
+            decrease_balance(state, source_validator, penalty)
+            initiate_validator_exit(state, pending_consolidation.source_index)
+            source_validator.slashed = add_flag(source_validator.slashed, SLASHED_PROPOSER_FLAG_INDEX)
+
+        if is_slashed_attester(target_validator):
+            if is_attester_slashable_validator(source_validator, current_epoch):
+                slash_validator(state, pending_consolidation.source_index)
+
+        return
+
+    # Don't move the balance if the source withdrew
+    if source_validator.withdrawable_epoch < current_epoch:
+        return
+
+    # Move active balance
+    active_balance_ceil = MIN_ACTIVATION_BALANCE if has_eth1_withdrawal_credential(source_validator) else MAX_EFFECTIVE_BALANCE
+    active_balance = min(state.balances[pending_consolidation.source_index], active_balance_ceil)
+    state.balances[target_index] += active_balance
+    # Excess balance above current active balance ceil will be withdrawn
+    state.balances[pending_consolidation.source_index] = state.balances[pending_consolidation.source_index] - active_balance
+
+    # Update target exit and withdrawable epochs if the target exited
+    if target_validator.exit_epoch != FAR_FUTURE_EPOCH:
+        target_validator.exit_epoch = compute_exit_epoch_and_update_churn(state, active_balance)
+        target_validator.withdrawable_epoch = Epoch(target_validator.exit_epoch + config.MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
+
+    # Balance is not exiting the active set, do not apply churn
+    if source_validator.exit_epoch > current_epoch:
+        source_validator.exit_epoch = current_epoch
```

Comment:
```suggestion
    source_validator.exit_epoch = min(source_validator.exit_epoch, compute_activation_exit_epoch(current_epoch))
```

Reply:
`compute_activation_exit_epoch` logic is already incorporated in `process_consolidation`:

https://github.com/michaelneuder/consensus-specs/pull/12/files#diff-1cdf2318cbed64bdc20bd25631130be090a72ca7ff4f05bbc74c0c63a634e0f1R2080